internal/clients/ec2/ec2_client.go

```diff
@@ -99,3 +99,16 @@ func (c *Client) CreateEC2ClientFromConfig(crd *stsTypes.Credentials) (*Client,
 
 	return newClient, nil
 }
+
+func (c *Client) ListInstanceTypes() ([]types.InstanceTypeInfo, error) {
+	log.Trace().Msg("Listing AWS EC2 instance types")
+	input := &ec2.DescribeInstanceTypesInput{
+		MaxResults: aws.Int32(100),
+	}
+	resp, err := c.ec2.DescribeInstanceTypes(c.context, input)
+	if err != nil {
+		return nil, fmt.Errorf("cannot list instance types: %w", err)
+	}
```

Comment:
So I am not sure if I am going to even receive an error if there are more than 100 records.

Reply:
My point was to actually return (own) error when the list is `len(n) == 100`